src/Microsoft.ML.Tokenizers/Model/Tiktoken.cs

```diff
@@ -296,122 +288,206 @@ public override IReadOnlyList<Token> Encode(string text, bool isSpecialToken = f
         /// <summary>
         /// Encode text to a list of Ids.
         /// </summary>
-        /// <param name="text">The text to encode. If the value of the parameter <paramref name="isSpecialToken"/> is true, the entire text will be treated as a special token.</param>
-        /// <param name="isSpecialToken">Specifies whether the entire <paramref name="text"/> is considered a special token.</param>
+        /// <param name="text">The text to encode.</param>
         /// <param name="accumulatedIds">The list of accumulated Ids.</param>
-        public override void EncodeToIds(ReadOnlySpan<char> text, bool isSpecialToken, IList<int> accumulatedIds)
+        /// <param name="textLength">The length of the text that encompasses the maximum encoded tokens.</param>
+        /// <param name="maxTokens">The maximum number of tokens to encode.</param>
+        /// <returns>The number of tokens that the input text will be encoded to.</returns>
+        public override int EncodeToIds(ReadOnlySpan<char> text, IList<int> accumulatedIds, out int textLength, int maxTokens = int.MaxValue)
         {
+            Debug.Assert(maxTokens > 0);
+
             if (text.IsEmpty)
             {
-                return;
+                textLength = 0;
+                return 0;
             }
 
-            if (isSpecialToken)
+            if (_cache.TryGetValue(text, out (int[] Bytes, string Token) tokenIds))
```

Comment:
Why is this called "Bytes"? Isn't this the IDs?

Reply:
I fixed that.